components/mpas-ocean/src/shared/mpas_ocn_vmix.F

```diff
@@ -1386,6 +1532,70 @@ subroutine ocn_vmix_init(domain, err)!{{{
    !--------------------------------------------------------------------
 
    end subroutine ocn_vmix_init!}}}
+
+!***********************************************************************
+!
+!  routine tridiagonal_solve
+!
+!> \brief   Solve the matrix equation Ax=r for x, where A is tridiagonal.
+!> \author  Mark Petersen
+!> \date    September 2011
+!> \details
+!>  Solve the matrix equation Ax=r for x, where A is tridiagonal.
+!>  A is an nxn matrix, with:
+!>  a sub-diagonal, filled from 1:n-1 (a(1) appears on row 2)
+!>  b diagonal, filled from 1:n
+!>  c sup-diagonal, filled from 1:n-1  (c(1) apears on row 1)
+!
+!-----------------------------------------------------------------------
+   subroutine tridiagonal_solve(a,b,c,r,x,n) !{{{
+
+      !-----------------------------------------------------------------
+      !
+      ! input variables
+      !
+      !-----------------------------------------------------------------
+
+      integer,intent(in) :: n
+      real (KIND=RKIND), dimension(n), intent(in) :: a,b,c,r
+
+      !-----------------------------------------------------------------
+      !
+      ! output variables
+      !
+      !-----------------------------------------------------------------
+
+      real (KIND=RKIND), dimension(n), intent(out) :: x
+
+      !-----------------------------------------------------------------
+      !
+      ! local variables
+      !
+      !-----------------------------------------------------------------
+
+      real (KIND=RKIND), dimension(n) :: bTemp,rTemp
+      real (KIND=RKIND) :: m
+      integer i
+
+      ! Use work variables for b and r
+      bTemp(1) = b(1)
+      rTemp(1) = r(1)
+
+      ! First pass: set the coefficients
+      do i = 2,n
+         m = a(i-1)/bTemp(i-1)
+         bTemp(i) = b(i) - m*c(i-1)
+         rTemp(i) = r(i) - m*rTemp(i-1)
+      end do
+
+      x(n) = rTemp(n)/bTemp(n)
+       ! Second pass: back-substition
+      do i = n-1, 1, -1
+         x(i) = (rTemp(i) - c(i)*x(i+1))/bTemp(i)
+      end do
+
+   end subroutine tridiagonal_solve !}}}
+
```

Comment:
I don't think this is a good addition.  I believe the performance team has inlined all the tri-diagonal solves and we probably shouldn't add it back.  But maybe since it won't be called in e3sm simulations for some time it's okay as is.  What do you think @philipwjones?

Reply:
Indeed. You can either use the other routines as a model or look to see if one of the other versions would work with some suitable modifications/pre-computation of input arguments.

Reply:
@scalandr looks like you got the function inlined.  If so I think you can remove this block of code.